app/controllers/courses_controller.rb

```diff
@@ -46,11 +46,7 @@ def update
 
   def destroy
     validate
-    @course.destroy
-    WikiCourseEdits.new(action: :update_course,
-                        course: @course,
-                        current_user: current_user,
-                        delete: true)
+    DeleteCourseWorker.schedule_edits(course: @course, current_user: current_user)
```

Comment:
Since this does more than just delete it, let's name the method differently. `schedule_deletion` perhaps?